compiler/rustc_hir/src/stable_hash_impls.rs

```diff
@@ -208,8 +208,13 @@ impl<'tcx, HirCtx: crate::HashStableContext> HashStable<HirCtx> for OwnerNodes<'
     fn hash_stable(&self, hcx: &mut HirCtx, hasher: &mut StableHasher) {
         // We ignore the `nodes` and `bodies` fields since these refer to information included in
         // `hash` which is hashed in the collector and used for the crate hash.
-        let OwnerNodes { hash_including_bodies, hash_without_bodies: _, nodes: _, bodies: _ } =
-            *self;
+        let OwnerNodes {
+            hash_including_bodies,
+            hash_without_bodies: _,
+            nodes: _,
+            bodies: _,
+            local_id_to_def_id: _,
```

Comment:
😆 this is your code, we just accidentally took it over and I didn't realize at all what was going on.

Since we require 

> x != y implies hash_stable(x) != hash_stable(y)

for stable hashing, but any info in `local_id_to_def_id` is dependent on the body, then just hashing the body satisfies this condition.

Reply:
I did not intend to congratulate myself. Sorry for the impoliteness.
I just thought it was not obvious that any info in `local_id_to_def_id` is dependent on the body, because the `LocalDefId`s do not appear inside HIR. I'll probably end up adding the comment in another PR, no need to block this one.

Reply:
Oops I thought we added the comment. Yea, let's do it in a follow up

cc @spastorino